src/NuGet.Core/NuGet.Protocol/Providers/ServiceIndexResourceV3Provider.cs

```diff
@@ -62,15 +62,12 @@ public override async Task<Tuple<bool, INuGetResource>> TryCreate(SourceReposito
                 if (!_cache.TryGetValue(url, out cacheInfo) ||
                     entryValidCutoff > cacheInfo.CachedTime)
                 {
-                    // Track if the semaphore needs to be released
-                    var release = false;
-                    try
-                    {
-                        await _semaphore.WaitAsync(token);
-                        release = true;
+                    await _semaphore.WaitAsync(token);
 
-                        token.ThrowIfCancellationRequested();
+                    token.ThrowIfCancellationRequested();
```

Comment:
I don't feel like this `token.ThrowIfCancellationRequested` is useful right after an async operation that uses the same token. I mean, it's probably not computationally expensive, so I doubt that we can measure any harm in checking the token, but if the cancellation was requested, the semaphore would have thrown.